tests/unit/htdocs/kernel/XoopsBlockPhpBlockTest.php

```
@@ -0,0 +1,730 @@
+    /**
+     * Verify that the welcome block returns HTML with username for logged-in users.
+     */
+    #[Test]
+    public function exampleWelcomeBlockReturnsHtmlForLoggedInUser(): void
+    {
+        $originalUser   = $GLOBALS['xoopsUser'] ?? null;
+        $originalConfig = $GLOBALS['xoopsConfig'] ?? null;
+
+        try {
+            require_once XOOPS_ROOT_PATH . '/kernel/user.php';
+            $user = new \XoopsUser();
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '561', column '25'). (undefined)

(MissingImport)